trio/_tools/gen_exports.py

```diff
@@ -36,7 +43,7 @@
 """
 
 
-def is_function(node):
+def is_function(node: ast.AST) -> TypeGuard[ast.FunctionDef | ast.AsyncFunctionDef]:
```

Comment:
style: Since there's a bunch of `ast.FunctionDef | ast.AsyncFunctionDef` I created a TypeAlias for it, and while it does make the signatures a bit easier to parse I'm not even sure myself it's worth it.